File: RB101_programming_foundations/lesson_5/5_practice_problems/ex_09.rb
# Given this data structure, return a new array of the same structure but with the sub arrays being ordered (alphabetically or numerically as appropriate) in descending order.

arr = [['b', 'c', 'a'], [2, 1, 3], ['blue', 'black', 'green']]

sorted_arr = arr.map do |array|
               array.sort { |a, b| b <=> a }
             end

File: JS101_programming_foundations/lesson_5/practice_problems/ex_09.js
// Given the following data structure, return a new array with the same structure, but with the subarrays ordered-- alphabetically or numerically as appropriate -- in ascending order.

let arr = [['b', 'c', 'a'], [2, 1, 3], ['blue', 'black', 'green']];

let newArr = arr.map(subarr => {
  if (typeof subarr[0] === 'number') {
    return subarr.slice().sort((a, b) => a - b);
  } else {
    return subarr.slice().sort();
  }
});


console.log(newArr);
console.log(arr);